src/containers/shared/components/DomainLink.tsx

```diff
@@ -9,12 +9,22 @@ export interface Props {
 
 const DomainLink = (props: Props) => {
   const { className, decode = false, domain } = props
+
+  // If decode is true, decode the domain
+  const decodedDomain = decode ? decodeHex(domain) : domain
+
+  // Check if the decodedDomain contains a protocol (http:// or https://)
+  const domainHasProtocol = decodedDomain.match(/^([a-z][a-z0-9+\-.]*):\/\//)
```

Comment:
nit: I think test might be faster and you should define the regex outside the component so it doesn't have to be compiled at every usage.